### PR TITLE
Fix windows installation problem (Issue #1634)

### PR DESCRIPTION
**Pull request by badpandabear**
As described in issue [1634](https://github.com/pypeit/PypeIt/issues/1634) code added to support OpenMP was failing on windows because gcc was not installed. Through testing I found that installing gcc was not sufficient to deal with this because setup tools would always use the Microsoft Visual Studio compiler when building the extension.

It turns out the [extension_helpers](https://github.com/astropy/extension-helpers) package, already used by PypeIt, had similar code to test for and use OpenMP that worked on windows. So I updated `bspline/setup_package.py` and `pypeit_c_enabled` to use that code instead.  I also updated the installation docs to provide more information on how to install under Windows and how to enable OpenMP on Macs.  I also referenced Windows Subsystem for Linux as an alternative for running under Windows.

### CODE REVIEW

File: doc/installing.rst [hunk 1]
```diff
@@ -522,7 +546,8 @@ GCC 12.x via ``homebrew``, you would get ``pypeit`` to use it by doing, for exam
 
 .. code-block:: console
 
-    CC=gcc-12 pip install pypeit
+    $ export CC=gcc-12
```
Comment (tepickering) [inline doc/installing.rst]: should be gcc-14 as of now. `/opt/homebrew/bin/gcc` is more general option for homebrew installs.

File: pypeit/bspline/setup_package.py [hunk 2]
```diff
@@ -55,16 +14,17 @@ def check_for_openmp():
...
+                          export_symbols=['bspline_model', 'solution_arrays',
+                                          'cholesky_band', 'cholesky_solve',
+                                          'intrv'])
+    
+    # extension_helpers will check for opnmp support by trying to build
+    # some test code with the appropriate flag. openmp provides a big performance boost, but some
+    # systems, notably apple's version of clang that xcode provides, don't support it out of the box.
+
+    add_openmp_flags_if_available(extension)
+    return [extension]
```
Comment (tepickering) [inline pypeit/bspline/setup_package.py]: looks like missing newline at end of file.

**Review by tepickering — APPROVED** [review]
i noted a couple changes to be done, most importantly update the version of `gcc` from 12 to 14. or just specify the homebrew gcc specifically. everything else looks good. thanks for the quick work!

File: doc/installing.rst (hunk 1 above)
Comment (badpandabear) [reply doc/installing.rst]: I went with the general /opt/homebrew/bin/gcc

File: pypeit/bspline/setup_package.py (hunk 2 above)
Comment (badpandabear) [reply pypeit/bspline/setup_package.py]: Fixed

**Review by kbwestfall — APPROVED** [review]
Looks great!  Thanks for the detailed update to the windows installation instructions!

**Pull request merged**